core/src/main/java/org/jruby/Main.java

```diff
@@ -531,6 +537,26 @@ protected static int handleRaiseException(RaiseException rj) {
         }
     }
 
+    public static void printTruffleTimeMetric(String id) {
+        if (Options.TRUFFLE_METRICS_TIME.load()) {
+            final long millis = System.currentTimeMillis();
+            System.err.printf("%s %d.%03d%n", id, millis / 1000, millis % 1000);
+        }
+    }
+
+    private static void printTruffleMemoryMetric() {
+        if (Options.TRUFFLE_METRICS_MEMORY_USED_ON_EXIT.load()) {
+            System.gc();
```

Comment:
JRuby has a great deal of static state (even though it is actually re-entrant) so I still need to check heap usage right at the end. Now I think about it I'm not sure what the point is of the GC at the end, given that I need to check heap usage after it anyway.